.github/workflows/docker-ci.yml

```diff
@@ -0,0 +1,628 @@
+        run: mkdir -p ./trivy-reports
+  
+      - name: Scan the Docker image with Trivy
+        uses: aquasecurity/trivy-action@master
+        with:
+          input: './image.tar'        
+          format: 'table'
+          output: './trivy-reports/trivy-report-${{ needs.PREPARE-job.outputs.dockerfile_name }}.txt'
+          severity: 'MEDIUM,CRITICAL,HIGH'
+      
+      - name: Add Trivy report to GitHub Actions summary
+        run: |
+          echo '## Trivy Scan Report for ${{ needs.PREPARE-job.outputs.dockerfile_name }}' >> $GITHUB_STEP_SUMMARY
+          cat ./trivy-reports/trivy-report-${{ needs.PREPARE-job.outputs.dockerfile_name }}.txt >> $GITHUB_STEP_SUMMARY
+
+      - name: Upload Trivy scan report
+        uses: actions/upload-artifact@v4
+        with:
+            name: trivy-report-${{ needs.PREPARE-job.outputs.dockerfile_name }}
+            path: ${{ github.workspace }}/trivy-reports/trivy-report-${{ needs.PREPARE-job.outputs.dockerfile_name}}.txt
+
+  PUSH-PRIV-job:
+    needs: [BUILD-job, PREPARE-job]
+    runs-on: ${{ needs.PREPARE-job.outputs.runner_label }}
+    if: needs.PREPARE-job.outputs.proceed_valid == 'true'
+    env:
+     BUCKET: ${{ vars.ACACIA_BUCKETNAME }} # BYO or pawsey0001-image-compilation if compile for project
+     DESTINATION_PATH: ${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/${{ needs.PREPARE-job.outputs.date }}
+    steps:
+      - name: Print hostname and set approved label for deployment
+        id: set_approved_label
+        run: |
+          echo "Hostname: $(hostname)"
+
+      - name: Copy back persistent storage "$HOME/runner/artifacts" to current directory in case of running before scan
+        run: |
+          if [ -f "$HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar" ]; then
+            echo "File already exists, skipping copy."
+          else
+            cp $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar ${GITHUB_WORKSPACE}
+            echo "Copied image.tar from $HOME/runner/artifacts/${{ needs.PREPARE-job.outputs.dockerfile_name }}-${{ needs.PREPARE-job.outputs.platform_tag }}/image.tar to ${GITHUB_WORKSPACE}"
+          fi
+
+      - name: Setup rclone
+        uses: ./.github/actions/setup-rclone
+        with:
+          access_key_id: ${{ secrets.ACACIA_ACCESS_KEY_ID }}
+          secret_access_key: ${{ secrets.ACACIA_SECRET_ACCESS_KEY }}
+          endpoint: https://projects.pawsey.org.au
+          bucket: ${{ env.BUCKET }}
+          destination_path: ${{ env.DESTINATION_PATH }}
+
+      - name: Upload image tar to S3 with rclone
+        run: |
+          set -e
+          rclone copy ${{ github.workspace }}/image.tar pawsey0001:"${{ env.BUCKET }}/${{ env.DESTINATION_PATH }}/"\
+              --transfers=16 \
+              --drive-chunk-size=128M \
```

Comment:
We might also want to set `--s3-chunk-size` and `--s3-upload-concurrency` and I'd almost be tempted to have these set by looking at the file we are uploading.

Reply:
Good suggestion. I have patched this part as follows. 
```
- name: Upload image tar to S3 with rclone
        run: |
          set -e
      
          # calculate file size
          FILE_SIZE=$(wc -c < "${{ github.workspace }}/image.tar")
          echo "File size: $FILE_SIZE bytes"
      
          # dynamically set rclone parameters based on file size: 500MB, 5G,others
          if [ "$FILE_SIZE" -lt $((1024 * 1024 * 500)) ]; then
            S3_CHUNK_SIZE="16M"
            S3_UPLOAD_CONCURRENCY=4
            MULTI_THREAD_STREAMS=2
          elif [ "$FILE_SIZE" -lt $((1024 * 1024 * 5000)) ]; then
            S3_CHUNK_SIZE="64M"
            S3_UPLOAD_CONCURRENCY=8
            MULTI_THREAD_STREAMS=4
          else
            S3_CHUNK_SIZE="128M"
            S3_UPLOAD_CONCURRENCY=16
            MULTI_THREAD_STREAMS=8
          fi
    
      
          echo "Using S3 chunk size: $S3_CHUNK_SIZE"
          echo "Using S3 upload concurrency: $S3_UPLOAD_CONCURRENCY"
          echo "Using multi-thread streams: $MULTI_THREAD_STREAMS"
```

Reply:
Do you think the parameters are reasonable? @pelahi